hyperbolic-rs/rudders/datasets/keen.py

```diff
@@ -64,3 +66,56 @@ def build_iid2title(item_id_key, item_title_key):
             if item_id_key in line and item_title_key in line:
                 iid2title[line[item_id_key]] = line[item_title_key][1:-1]
     return iid2title
+
+
+def get_keens(data):
+    all_keens = {}
+    for item in data:
+        keen_id = item["keen_id"]
+        gem = None
+        if "gem_id" in item:
+            gem = Gem(item)
+
+        if keen_id in all_keens:
+            keen = all_keens[keen_id]
+        else:
+            keen = Keen(item["keen_id"], item["keen_title"], get_value(item, "keen_description"),
+                        item["keen_creator_uid"])
+            all_keens[keen_id] = keen
+
+        if gem is not None and not gem.is_empty():
+            keen.gems.append(gem)
+    return all_keens
+
+
+def process_input(string):
+    string = string[1:-1] if type(string) == str else ""
```

Comment:
Add a comment to explain that the dataset contains extra quote characters at the start and end, which is why you remove the first and last character.

Reply:
Done